thanos/templates/compact-deployment.yaml

```
@@ -79,4 +79,8 @@ spec:
             - key: google.json
               path: google.json
     {{- end }}
+    {{- with .Values.compact.securityContext }}
+      securityContext:
+{{ toYaml . | indent 8 }}
```

This might be more readable using `nindent`.
See: http://masterminds.github.io/sprig/strings.html and https://github.com/helm/helm/blob/master/docs/charts_tips_and_tricks.md#using-the-include-function
```suggestion
        {{ toYaml . | nindent 8 }}
```